sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/attestation/attestation_administration_client.hpp"
+#include "azure/attestation/attestation_client.hpp"
+#include "private/attestation_client_models_private.hpp"
+#include "private/attestation_client_private.hpp"
+#include "private/attestation_common_request.hpp"
+#include "private/attestation_deserializers_private.hpp"
+#include "private/package_version.hpp"
+#include <azure/core/base64.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/diagnostics/log.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+#include <shared_mutex>
+
+#include <string>
+
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::Models;
+using namespace Azure::Security::Attestation::_detail;
+using namespace Azure::Security::Attestation::Models::_detail;
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_internal;
+using namespace Azure::Core::Http::_internal;
+using namespace Azure::Core::Diagnostics::_internal;
+using namespace Azure::Core::Diagnostics;
+
+namespace Azure { namespace Security { namespace Attestation {
+  const Models::AttestationType AttestationType::SgxEnclave("SgxEnclave");
+  const Models::AttestationType AttestationType::OpenEnclave("OpenEnclave");
+  const Models::AttestationType AttestationType::Tpm("Tpm");
+  const Models::PolicyModification PolicyModification::Removed("Removed");
+  const Models::PolicyModification PolicyModification::Updated("Updated");
+
+  AttestationAdministrationClient::AttestationAdministrationClient(
+      std::string const& endpoint,
+      std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+      AttestationAdministrationClientOptions const& options)
+      : m_endpoint(endpoint), m_apiVersion(options.Version.ToString()),
+        m_tokenValidationOptions(options.TokenValidationOptions)
+  {
+    std::vector<std::unique_ptr<HttpPolicy>> perRetrypolicies;
+    if (credential)
+    {
+      m_credentials = credential;
+      Azure::Core::Credentials::TokenRequestContext const tokenContext
+          = {{"https://attest.azure.net/.default"}};
+
+      perRetrypolicies.emplace_back(
+          std::make_unique<BearerTokenAuthenticationPolicy>(credential, tokenContext));
+    }
+    m_apiVersion = options.Version.ToString();
+    std::vector<std::unique_ptr<HttpPolicy>> perCallpolicies;
+
+    m_pipeline = std::make_shared<Azure::Core::Http::_internal::HttpPipeline>(
+        options,
+        "Attestation",
```

Comment:
mmm, since we are using the same service package for admin and non-admin, telemetry won't make a difference between them right?
At least you change this to `Attestation-admin`.  But I am not sure.

Reply:
Telemetry shouldn't really distinguish between them. Certainly not on the service side it isn't distinguished.